tests/testbed_setup/test_xr_migration.py

```diff
@@ -0,0 +1,210 @@
+import logging
+from time import sleep
+import pytest
+import re
+
+
+logger = logging.getLogger(__name__)
+
+pytestmark = [
+    pytest.mark.topology('any'),
+]
+
+
+mig_script = "sonic_migration_xr.py"
+xr_user = ""
+xr_pass = ""
+xr_prompt = "RP/0/RP0/CPU0:.*#"
+mgmt_int = "Mg0/RP0/CPU0/0"
+mgmt_ip_mask = "10.250.0.20 255.255.255.0"
+mgmt_gw = "10.250.0.1"
+rollback_file_path = "/home/cisco/Secureboot/Rollback-Files/"  # File paths must have trailing slash
+mig_file_path = "/home/cisco/Secureboot/Migration-Files/"
+scp_user = ""
+scp_pass = ""
+scp_host = "10.250.0.245"
+
+
+def read_con(duthost_console, prompt):
+    output = ''
+    logger.debug(f"reading console until: {prompt}")
+    while True:
+        try:
+            line = duthost_console.read_until_pattern(duthost_console.RETURN)
+            logger.debug(line)
+            output = output + line
+            found = re.search(prompt, line)
+            if found:
+                break
+        except Exception:
+            continue
+    return output
+
+
+def test_xr_migration(duthost_console, duthosts, enum_supervisor_dut_hostname, creds):
+    duthost = duthosts[enum_supervisor_dut_hostname]
```

Comment:
This is only supported on Cisco chassis I suppose?
Is there any logic to skip this test case on other platforms?

Reply:
This test installs a Cisco proprietary image that will only run on Cisco hardware so only Cisco platforms are supported.